src/main/java/bio/terra/app/model/CloudResource.java

```diff
@@ -0,0 +1,60 @@
+package bio.terra.app.model;
+
+import bio.terra.common.EnumUtils;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.exc.InvalidFormatException;
+
+import java.io.IOException;
+
+/**
+ * Represents resource types across different cloud providers
+ */
+@JsonDeserialize(using = CloudResource.CloudResourceDeserializer.class)
+public interface CloudResource {
+    String name();
+
+    class CloudResourceDeserializer extends StdDeserializer<CloudResource> {
+
+        public CloudResourceDeserializer() {
+            this(CloudResource.class);
+        }
+        public CloudResourceDeserializer(Class<?> vc) {
+            super(vc);
+        }
+
+        @Override
+        public CloudResource deserialize(
+            JsonParser jsonParser,
+            DeserializationContext ctxt) throws IOException {
+            JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+            if (node.isNull()) {
+                return null;
+            }
+            if (!node.isTextual()) {
+                throw new InvalidFormatException(
+                    jsonParser,
+                    "Invalid representation of CloudResource",
+                    node,
+                    CloudResource.class);
+            }
+            String value = node.textValue();
+            CloudResource resource = EnumUtils.valueOfLenient(GoogleCloudResource.class, value);
```

Comment:
I would expect the class type to come last, as in `ObjectMapper.readValue()`. Also if you have this helper method, do we also need `AzureRegion.fromName()`?

Reply:
I can switch the order.  I'm not sure I follow about the `AzureRegion.fromName`?

Reply:
It looked to me like `AzureRegion.fromName(value)` was the same as `EnumUtils.valueOfLenient(AzureRegion.class, value)`?